### DISCUSSION TITLE
Indexed Image Names Differ From Original File Names

### DISCUSSION THREAD

**Discussion by cfis** [photoprism/photoprism#4062] [Q&A]
#### 1. What is not working as documented?

When indexing files from the original folder, my expectation is that the name showed for the image under the search menu matches the original file name. Most of the time that is true.

But sometimes it is not. The indexed image name can be "Unknown" or "Portrait" or "Moment" or other things. So it seems Photoprism has some logic for figuring out an image name that is different than the file name. 

#### 2. How can we reproduce it?

Not sure.

#### 3. What behavior do you expect?
That the image name is the same as the file name. Having a bunch of images with the name "Portrait" isn't very helpful :(

#### 6. Which software versions do you use?

[PhotoPrism® Plus] Build [231128-f48ff16ef]
Fedora 39 host, K8 cluster (k0s)

(b) Database Type & Version: MariaDB, MySQL, SQLite,...

MariaDB

**Comment by cfis** [comment]
Looking at this a bit more:

```
select photo_name, photo_title
from photos
```

I can see the the photo_name is the file name, while the photo title is variable. In some cases, the title is copied from the exif Title tag, so that of course makes sense.

But in cases without an exif tag its not quite so clear. One pattern I see is:

Name        Title
-----------------
Photo 1 -> Photo
Photo 2 -> Photo
Photo 3 -> Photo
Photo 4 -> Photo
Photo 5 -> Photo

So you end up with 5 photos with the same name, but they originally had different file names. Photos with just numbers as names get some random titles.

Maybe it is just me, but I find this surprising and not very helpful. So although this is likely a terrible idea (breaks redindexing maybe?), I just ran this sql query:

```
update photos
set photo_title = photo_name;
```

**Reply by graciousgrey** [reply]
The photo title is not intended to be the file name. It is created on the basis of various information, e.g. title information from Exif or xmp metadata. If these are not available, the title can also consist of parts of the file name or recognized labels, people or places. You can find some examples at https://demo.photoprism.app/library/browse.
The titles are also updated automatically when new information is available.

It is planned for the future to make the information displayed in the card view configurable so that people who prefer other information than the generated title can have it displayed. 

Changing the photo title is also possible via the web interface, so there should be no problems if you change it to a valid value. Just make sure that you set the title_src field to "manual" so that your changes are not overwritten later.

**Comment by cfis** [comment]
Having an option to show the filename instead of title when listing images would be great and solves the problem.

Updating them manually really isn't workable since I have 10,000 images I added.

**Reply by lastzero** [reply]
You can currently sort by file name for this or go to Library > Originals.

**Reply by cfis** [reply]
Sorting by file name doesn't help - I just see a bunch of pictures with the same name. And Library -> Originals only shows 999 photos.

A toggle to display by file name instead of title would be a great addition.

**Reply by lastzero** [reply]
Again, TITLE is not the same as FILE NAME. See the following screenshot for how the FILE NAME is displayed as previously described:

![IMG_2920](https://github.com/photoprism/photoprism/assets/301686/5cf17e4e-dc85-4185-b67d-58e8279bbd01)

**Comment by cfis** [comment]
Right I understand that title is not the same as file name. At least in my case, I would much prefer seeing the file name at the top. In addition, when you click on the image, to see the file name at the bottom and not the title.

Perhaps it is just me - but the images I have do not have useful titles and the auto generated ones by photoprism convey less information that the file name. To make this concrete, say I have pictures Robert 1, Robert 2, Steve 1, Steve 2, Alice 1, Alice 2, all taken on a vacation. When they get indexed by Photoprism, the titles become "Portrait" for all of them. So if you scroll through the images one at a time in full screen mode you see "Portrait" and then "Portrait" and then "Portrait" etc. Or another example, every photo from my Grand Canyon trip is now titled "Coconio County."

Of course I could run exiftool on them and mass update the titles to match the file names and then reimport. Or run the sql query I did.

I do think showing titles is the right default. I imagine most people are adding files taken by their phones, where the file name is not useful and trying to figure out a title is a great idea.

But it would be nice to have an option to emphasize file name over title. Or maybe an option to control on indexing how the Title is generated? If no that, then it would be nice to have a setting to always show the file name. Right now it, like in your screenshot, it only shows if you sort by filename and not other ways such as date.

**Reply by DovieW** [reply]
I 100% agree with you @cfis. My titles for my files are useless. I want the filename front and center I don't care about anything else
